extensions/amp-story/1.0/amp-story-grid-layer.js

```diff
@@ -104,6 +109,23 @@ export class AmpStoryGridLayer extends AmpStoryBaseLayer {
     return this.isFirstPage_;
   }
 
+  /**
+   * Returns true if the page is in the hashString
+   * @return {boolean}
+   */
+  isPrerenderActivePage() {
+    if (this.isActivePage_ === null) {
+      const hashId = parseQueryString(this.win.location.href)['page'];
+      const identSelector = `amp-story-page#${escapeCssSelectorIdent(
+        hashId
+      )} amp-story-grid-layer`;
+      this.isActivePage_ = hashId
+        ? matches(this.element, identSelector)
+        : this.isFirstPage();
+    }
+    return this.isActivePage_;
```

Comment:
I don't know why my comment didn't send the first time, let me type it again.

You're making this code much harder to follow than it should be. I know it only runs one query but you have two, and the second one is called by the fallback of a ternary condition.
Instead of just adding an extra "if" to existing code, sometimes it's easy and better to re-write a small portion and make things easy.

Here, we could have one single method with one single query:

```
isPrerenderActivePage_() {
  // early return if method already executed

  let selector = 'amp-story-page:first-of-type';

  // Support prerender with branching .
  const hashId = ...;
  if (hashId) {
    selector += `, amp-story-page#${hashId}`;
  }

  return matches(this.element, selector);
}
```

It's less lines, less methods, less `if`, so much easier to test, maintain, and understand for a new developer.